### PR TITLE
do not use fatalError in Triple.getHostTriple

### PR DESCRIPTION
**Pull request by tomerd**
motivation: avoid crashes, use structured error handling

changes:
* make Triple.getHostTriple throwable and remove use of fatalError
* update call sites

rdar://90317112

### CODE REVIEW

**Comment by tomerd** [comment]
@swift-ci smoke test

File: Tests/BuildTests/MockBuildTestHelper.swift [hunk 1]
```diff
@@ -76,7 +76,7 @@ func mockBuildParameters(
     useExplicitModuleBuild: Bool = false,
     linkerDeadStrip: Bool = true
 ) -> BuildParameters {
-    return BuildParameters(
+    return try! BuildParameters(
```
Comment (tomerd) [inline Tests/BuildTests/MockBuildTestHelper.swift:79]: fine in tests

**Comment by tomerd** [comment]
https://github.com/apple/sourcekit-lsp/pull/721

@swift-ci smoke test

**Review by MaxDesiatov — APPROVED** [review]
This is great, thank you!

**Pull request merged**